jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -341,64 +320,51 @@ ecma_builtin_object_object_get_own_property_symbols (ecma_value_t this_arg, /**<
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_object_object_seal (ecma_value_t this_arg, /**< 'this' argument */
-                                 ecma_value_t arg) /**< routine's argument */
+ecma_builtin_object_object_seal (ecma_object_t *obj_p) /**< routine's argument */
 {
-  JERRY_UNUSED (this_arg);
   ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  ecma_collection_header_t *props_p = ecma_op_object_get_property_names (obj_p, ECMA_LIST_NO_OPTS);
 
-  /* 1. */
-  if (!ecma_is_value_object (arg))
-  {
-    ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Argument is not an object."));
-  }
-  else
-  {
-    /* 2. */
-    ecma_object_t *obj_p = ecma_get_object_from_value (arg);
+  ecma_value_t *ecma_value_p = ecma_collection_iterator_init (props_p);
 
-    ecma_collection_header_t *props_p = ecma_op_object_get_property_names (obj_p, ECMA_LIST_NO_OPTS);
+  while (ecma_value_p != NULL && ecma_is_value_empty (ret_value))
+  {
+    ecma_string_t *property_name_p = ecma_get_string_from_value (*ecma_value_p);
+    ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
 
-    ecma_value_t *ecma_value_p = ecma_collection_iterator_init (props_p);
+    /* 2.a */
+    ecma_property_descriptor_t prop_desc;
 
-    while (ecma_value_p != NULL && ecma_is_value_empty (ret_value))
+    if (!ecma_op_object_get_own_property_descriptor (obj_p, property_name_p, &prop_desc))
     {
-      ecma_string_t *property_name_p = ecma_get_string_from_value (*ecma_value_p);
-      ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
-
-      /* 2.a */
-      ecma_property_descriptor_t prop_desc;
-
-      if (!ecma_op_object_get_own_property_descriptor (obj_p, property_name_p, &prop_desc))
-      {
-        continue;
-      }
+      continue;
+    }
 
-      /* 2.b */
-      prop_desc.is_configurable = false;
+    /* 2.b */
+    prop_desc.is_configurable = false;
 
-      /* 2.c */
-      ECMA_TRY_CATCH (define_own_prop_ret,
-                      ecma_op_object_define_own_property (obj_p,
-                                                          property_name_p,
-                                                          &prop_desc,
-                                                          true),
-                      ret_value);
-      ECMA_FINALIZE (define_own_prop_ret);
+    /* 2.c */
+    ECMA_TRY_CATCH (define_own_prop_ret,
+                    ecma_op_object_define_own_property (obj_p,
+                                                        property_name_p,
+                                                        &prop_desc,
+                                                        true),
+                    ret_value);
+    ECMA_FINALIZE (define_own_prop_ret);
```

Comment:
Since we're already touching these functions, would you mind also removing these `ECMA_TRY_CATCH` blocks, that are immediately finalized? There are a few scattered around.

Reply:
@dbatyai I've updated the PR. Now the whole file is `ECMA_TRY_CATCH` free.